### PR TITLE
feat(typescript_indexer): add support for utf-16 offset lookup

### PR DESCRIPTION
**Pull request by zzmp**
Expands utf8 OffsetTable to allow lookup of UTF-16 offsets from UTF-8 offsets.
Exposes OffsetTable through the IndexerHost so that Plugins have access.

### CODE REVIEW

**Comment by zzmp** [comment]
cc @evmar 
cc @ayazhafiz

File: kythe/typescript/indexer.ts [hunk 1]
```diff
@@ -172,6 +181,16 @@ class StandardIndexerContext implements IndexerHost {
     this.rootDirs = rootDirs;
   }
 
+  getOffsetTable(path: string): utf8.OffsetTable {
```
Comment (ayazhafiz) [inline kythe/typescript/indexer.ts]:
```suggestion
  getOffsetTable(path: string): Readonly<utf8.OffsetTable> {
```

**Comment by zzmp** [comment]
recheck CLA, please.

**Comment by ayazhafiz** [comment]
@googlebot I consent.

**Comment by zzmp** [comment]
@googlebot check consent.

**Comment by ayazhafiz** [comment]
> @googlebot check consent.

Feel free bump whatever changes I added that caused this, I don't know why it's doing this.

File: kythe/typescript/utf8.ts [hunk 2]
```diff
@@ -116,22 +116,49 @@ export class OffsetTable {
...
+    let u16 = Infinity;
+    let byte = Infinity;
+    let span =
+        Math.min(Math.floor(findOfs / this.spanSize), this.offsets.length - 1);
+    // We may have overshot it, because the span was chosen from the UTF-16
+    // offset. If necessary, backtrack.
+    while (byte > findOfs) {
+      const offset = this.offsets[span--];
+      u16 = offset[0];
+      byte = offset[1];
```
Comment (evmar) [inline kythe/typescript/utf8.ts]: Perhaps

[u16, byte] = offset;

**Comment by zzmp** [comment]
Fixed. Ready to merge once the checks pass.

**Pull request merged**